### DISCUSSION TITLE
Oqtane with Syncfusion on .net6 and 3.03

### DISCUSSION THREAD

**Discussion by DrtyByt** [oqtane/oqtane.framework#2024] [Q&A] [answered]
Has anyone tried and/or been able to get Syncfusion to work with Oqtane in .net6 with 3.03?
I followed the blog that was on blazorhelpwebsite (great info source and guide). But I am having an issue. I can get the module to work, but it has no formatting. Looks like the css file isn't being used. I asked on the blazorhelpwebsite, but there was one response, but they don't have time to really look into it. 
I'll try some other stuff to see if I can get it to work, but i'm pretty much grasping at straws at this point.
![image](https://user-images.githubusercontent.com/4389397/155461381-33e90863-a4cf-4f41-8f9d-cd95d0376377.png)

**Comment by leigh-pointer** [comment]
@DrtyByt check out https://github.com/ADefWebserver/Oqtane.SyncfusionHelpDesk from @ADefWebserver  (Michael Washington)

**Reply by ADefWebserver** [reply]
@leigh-pointer - @DrtyByt already reached out to me. I am unsure why it is not working. I suggested this more recent tutorial instead: [Oqtane Module Creator](https://blazorhelpwebsite.com/ViewBlogPost/4)

**Reply by DrtyByt** [reply]
I've tried reviewing both and so far haven't found anything out of place. I think the issue comes from something that changed recently either in .net6 or with the 3.0+ version. 
The Helpdesk module was built in .net5 with a 2.x version.
The module creator is a great tutorial, but doesn't cover the content around the issue I'm seeing as it comes from the need for Syncfusion to have the css and javascript injected in a special way. Since the module gets data just not formatting I've been trying to look into what might be happening on how the css is injected, but I haven't had time to track that path all the way yet and see what might be going on. That is why I reached out in case someone could help me understand what might have changed either with .net6 or the 3.x version with the injection of the css (which I believe is the issue when checking dev tools I see the style tags, but I dont' see any HTML coming out of the style tags being defined).

**Reply by ADefWebserver** [reply]
@DrtyByt Can you see if this works: [Using Radzen In Oqtane Modules](https://blazorhelpwebsite.com/ViewBlogPost/46)? If it doesn't then there may be a bug in the Oqtane base.

**Reply by DrtyByt** [reply]
I tried, but didn't have much luck. I've been getting exceptions after installing the module. Going to run it in a clean Oqtane install when I have a chance, but that won't be till Monday.

**Reply by DrtyByt** [reply]
I wanted to try from a completely fresh build and had some time today. I'm getting an error after installing the module and restarting. I tried doing a rebuild of the TestExternal module to make sure I wasn't missing something from the build, but no luck.  see below, looks like it just isn't getting the Radzen assembly. So this is a completely different issue than what I'm having with Syncfusion. I'll dig into this a little but I didn't want to muck with the code as I was trying to do the test if this would work so you could find out if anything was amiss, but to me since right now it's a different issue I am not sure where the problem lays.

System.IO.FileNotFoundException
  HResult=0x80070002
  Message=Could not load file or assembly 'Radzen.Blazor, Version=2.18.8.0, Culture=neutral, PublicKeyToken=null'. The system cannot find the file specified.
  Source=MyCompany.TestExternal.Server.Oqtane
  StackTrace:
   at MyCompany.TestExternal.Server.Startup.ServerStartup.ConfigureServices(IServiceCollection services) in C:\TEMP\MyCompany.TestExternal\Server\Startup\ServerStartup.cs:line 18
   at Microsoft.Extensions.DependencyInjection.OqtaneServiceCollectionExtensions.<>c__DisplayClass9_0.<AddOqtaneServices>b__0(IServerStartup x) in C:\Users\kelem\OneDrive\Documents\GitHub\oq.test\Oqtane.Server\Extensions\OqtaneServiceCollectionExtensions.cs:line 229
   at System.Collections.Generic.List`1.ForEach(Action`1 action)
   at Microsoft.Extensions.DependencyInjection.OqtaneServiceCollectionExtensions.AddOqtaneServices(IServiceCollection services) in C:\Users\kelem\OneDrive\Documents\GitHub\oq.test\Oqtane.Server\Extensions\OqtaneServiceCollectionExtensions.cs:line 227
   at Microsoft.Extensions.DependencyInjection.OqtaneServiceCollectionExtensions.AddOqtane(IServiceCollection services, String[] supportedCultures) in C:\Users\kelem\OneDrive\Documents\GitHub\oq.test\Oqtane.Server\Extensions\OqtaneServiceCollectionExtensions.cs:line 31
   at Oqtane.Startup.ConfigureServices(IServiceCollection services) in C:\Users\kelem\OneDrive\Documents\GitHub\oq.test\Oqtane.Server\Startup.cs:line 116
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Span`1& arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Microsoft.AspNetCore.Hosting.ConfigureServicesBuilder.InvokeCore(Object instance, IServiceCollection services)
   at Microsoft.AspNetCore.Hosting.ConfigureServicesBuilder.<>c__DisplayClass9_0.<Invoke>g__Startup|0(IServiceCollection serviceCollection)
   at Microsoft.AspNetCore.Hosting.StartupLoader.ConfigureServicesDelegateBuilder`1.<>c__DisplayClass15_0.<BuildStartupServicesFilterPipeline>g__RunPipeline|0(IServiceCollection services)
   at Microsoft.AspNetCore.Hosting.ConfigureServicesBuilder.Invoke(Object instance, IServiceCollection services)
   at Microsoft.AspNetCore.Hosting.ConfigureServicesBuilder.<>c__DisplayClass8_0.<Build>b__0(IServiceCollection services)
   at Microsoft.AspNetCore.Hosting.StartupLoader.ConfigureServicesDelegateBuilder`1.<>c__DisplayClass14_0.<ConfigureServices>g__ConfigureServicesWithContainerConfiguration|0(IServiceCollection services)
   at Microsoft.AspNetCore.Hosting.ConventionBasedStartup.ConfigureServices(IServiceCollection services)

**Reply by ADefWebserver** [reply]
Yeah, all those samples worked in previous versions of Oqtane. Unfortunately, I don't have the bandwidth to look at this.

**Reply by ADefWebserver** [reply]
@sbwalker - Any reason why these projects would no longer work?
[MyCompany.TestExternal.zip](https://github.com/oqtane/oqtane.framework/files/8150037/MyCompany.TestExternal.zip)
[Uploading Syncfusion_Oqtane_Module.zip…]()

**Reply by sbwalker** [reply]
@DrtyByt the error message says that Oqtane "Could not load file or assembly 'Radzen.Blazor, Version=2.18.8.0" - have you verified that the Radzen.Blazor.dll is in the Oqtane.Server\bin\Debug\net6.0 folder? @ADefWebserver as part of the migration from .NET 5 to .NET 6 the /bin folder location has changed from net5.0 to net6.0 - which is important if you are running Oqtane within Visual Studio.

**Reply by ADefWebserver** [reply]
@sbwalker - Thanks for that. I did update the [Oqtane Module Creator ](https://blazorhelpwebsite.com/ViewBlogPost/4) tutorial to use that folder 

![image](https://user-images.githubusercontent.com/1857799/156239760-07182907-0b12-4bae-a7b1-6bc0d1a59504.png)

**Reply by ADefWebserver** [reply]
@sbwalker - I update the [Radzen](https://blazorhelpwebsite.com/ViewBlogPost/46) and [Syncfusion](https://blazorhelpwebsite.com/ViewBlogPost/47) blogs posts too.

**Comment by sbwalker** [comment, marked as answer]
@DrtyByt this issue is resolved in #2045. You can also resolve it in the current version by registering your resources in your module using the Declaration = ResourceDeclaration.Global attribute:

```
    public class HostResources : IHostResources
    {
        public List<Resource> Resources => new List<Resource>()
        {
            new Resource {
                ResourceType = ResourceType.Stylesheet,
                Declaration = ResourceDeclaration.Global,
                Url = "_content/Radzen.Blazor/css/default.css" }
        };
    }
```

That being said, I do not believe that the IHostResources approach is the recommended way to register static resources - it was more of a temporary solution while a more robust solution was being developed for Oqtane. The problem with IHostResources is that it registers the assets globally ie. it includes the stylesheet and JavaScript references on every page for every site within your multi-tenant installation - which is generally not a good idea. Declaring the resources locally in a module is much better as it means they are lazy loaded as needed. An example of a local declaration is in your Index.razor file:

```
@code {
    public override SecurityAccessLevel SecurityAccessLevel => SecurityAccessLevel.Anonymous;

    public override List<Resource> Resources => new List<Resource>()
    {
        new Resource { ResourceType = ResourceType.Stylesheet, Url = "_content/Radzen.Blazor/css/default.css" }
    };
```

**Reply by DrtyByt** [reply]
Thansk @sbwalker, I've been out of it for about a week as my kids brought home a nasty bug (not covid). Pretty much back to normal now. I'll try out the update to the razor pages themselves instead of doing the global declaration and see if that works, if not I'll swap over to the global declarataion. I'll let you know what I find. Thanks for putting in the fix as well!

**Reply by DrtyByt** [reply]
That worked perfectly. I've registered it into the Index.razor instead of IHost and no issues. Thanks @sbwalker

**Comment by markagama** [comment]
@DrtyByt are you still using syncfusion with oqtane? I've also tried to get the example working and cannot, two problems faced that seem insurmountable. I could do with some help and can pay if necessary.

**Comment by sbwalker** [comment]
@markagama the IHostResources interface was deprecated in Version 4.0. Please see this blog post: https://www.oqtane.org/blog/!/61/css-js-resources

**Reply by markagama** [reply]
thx, it appears that the example I followed is quite out of date. I'll make this change and see what happens.

**Comment by allschu** [comment]
This still works perfectly in version 5.2.1. :-)

**Reply by oqtanecms** [reply]
Has anyone been able to get Syncfusion SfToast to work with Oqtane? It seems that SfToast content cannot be shown in oqtane module.I tested SfToast with oqtane 5.21,Syncfusion.Blazor is 26.1.35.
any help you provide is greatly appreciated.
![SfToast](https://github.com/user-attachments/assets/240dcd3d-b273-4e8c-a0f0-8da41cfae565)

**Comment by ADefWebserver** [comment]
I updated https://github.com/ADefWebserver/Oqtane.SyncfusionHelpDesk
